arroyo-controller/src/states/scheduling.rs

```diff
@@ -350,9 +350,16 @@ impl State for Scheduling {
             needs_commits,
         }) = checkpoint_info.clone()
         {
-            let mut metadata = StateBackend::load_checkpoint_metadata(&ctx.config.id, epoch)
+            let mut metadata = match StateBackend::load_checkpoint_metadata(&ctx.config.id, epoch)
```

Comment:
Nit: using `ok_or_else` and `?` is generally preferable to a match statement where you're just trying to map the error case.